modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AutoDownloadSettingsFragment.kt

```diff
@@ -238,6 +238,9 @@ class AutoDownloadSettingsFragment :
 
     private fun handleNewEpisodesToggle(isEnabled: Boolean) {
         lifecycleScope.launch {
+            if (!isEnabled) {
+                async(Dispatchers.IO) { podcastManager.updateAllAutoDownloadStatus(Podcast.AUTO_DOWNLOAD_OFF) }.await()
```

Comment:
This is the only place that calls `updateAllAutoDownloadStatus` so we should convert the DAO method to a suspend function and let Room handle the context.

Reply:
I've refactor this method: [b72405c](https://github.com/Automattic/pocket-casts-android/pull/3166/commits/b72405c68c6ec72db851f1ade7bd04315446ea26)